ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -581,7 +627,10 @@ private JoinNode buildJoin(final Join root, final String prefix, final boolean i
     );
   }
 
-  private void verifyJoin(final JoinInfo joinInfo,
+  /**
+   * @return whether this is a foreign key join or not
```

Comment:
I don't love this but it seems better to have this method both perform verification and detect whether the join is an FK join or not than to have a second method duplicate the work in this one. I considered adding a join type enum rather than simply returning a boolean but we already have JoinType defined to mean something else: https://github.com/confluentinc/ksql/blob/89c1588440c26b6467ad58026ec664fbd0b6b85b/ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java#L69-L71

Open to suggestions for how to make this feel less hacky.

Reply:
IMHO, it make actually a lot of sense to do is this way. We don't need to over-engineer for now and a boolean flag is just fine. After some more pending refactoring, we might get into a cleaner state. For not it's good as-is IMHO.